src/buffer.rs

```diff
@@ -660,6 +673,7 @@ impl Buffer {
         if let Some(path) = &self.path {
             path.write(&self.content())?;
             self.dirty = false;
+            self.system_opened = false;
```

Comment:
Can you add a test case to test this path? Or more generically, the scenario `User saving a system-opened file will cause the file to appear in the buffer list`?